connectors/file/translator-parquet/src/main/java/org/teiid/translator/parquet/BaseParquetExecution.java

```diff
@@ -0,0 +1,341 @@
+import org.teiid.file.VirtualFile;
+import org.teiid.file.VirtualFileConnection;
+import org.teiid.language.Comparison;
+import org.teiid.language.LanguageObject;
+import org.teiid.language.Literal;
+import org.teiid.metadata.RuntimeMetadata;
+import org.teiid.translator.DataNotAvailableException;
+import org.teiid.translator.Execution;
+import org.teiid.translator.ExecutionContext;
+import org.teiid.translator.TranslatorException;
+
+
+public class BaseParquetExecution implements Execution {
+    @SuppressWarnings("unused")
+    protected ExecutionContext executionContext;
+    @SuppressWarnings("unused")
+    protected RuntimeMetadata metadata;
+    protected VirtualFileConnection connection;
+    protected boolean immutable;
+
+    // Execution state
+    protected ParquetQueryVisitor visitor = new ParquetQueryVisitor();
+    private VirtualFile[] parquetFiles;
+    private AtomicInteger fileCount = new AtomicInteger();
+    private ParquetFileReader reader;
+    private MessageType schema;
+    private MessageType filteredSchema;
+    private RecordReader<Group> rowIterator;
+    private MessageColumnIO columnIO;
+    private long pageRowCount;
+    protected HashMap<String, String> partitionedColumnsValue;
+    protected HashMap<String, Integer> partitionedColumnsHm = new HashMap<>();
+
+    public BaseParquetExecution(ExecutionContext executionContext,
+                                RuntimeMetadata metadata, VirtualFileConnection connection, boolean immutable) {
+        this.executionContext = executionContext;
+        this.metadata = metadata;
+        this.connection = connection;
+        this.immutable = immutable;
+    }
+
+    public void visit(LanguageObject command) throws TranslatorException {
+        this.visitor.visitNode(command);
+
+        if (!visitor.exceptions.isEmpty()) {
+            throw visitor.exceptions.get(0);
+        }
+    }
+
+    @Override
+    public void execute() throws TranslatorException {
+        String path = this.visitor.getParquetPath();
+        if(this.visitor.getTable().getProperty(ParquetMetadataProcessor.PARTITIONED_COLUMNS) != null) {
+            path += getDirectoryPath(this.visitor.getColumnPredicates(), this.visitor.getTable().getProperty(ParquetMetadataProcessor.PARTITIONED_COLUMNS));
+        }
+        this.parquetFiles = VirtualFileConnection.Util.getFiles(path, this.connection, true, false);
+        VirtualFile nextParquetFile = getNextParquetFile();
+        if (nextParquetFile != null) {
+            this.rowIterator = readParquetFile(nextParquetFile);
+        }
+    }
+
+    private String getDirectoryPath(HashMap<String, Comparison> predicates, String partitionedColumns) throws TranslatorException {
+        StringBuilder path = new StringBuilder(); // because we are only supporting the equality comparison as of now, otherwise we would return a list of paths to iterate over
+        String[] partitionedColumnsArray = getPartitionedColumns(partitionedColumns);
```

Comment:
A map with a constant value is effectively the same as a set.  You can change the type of partitionedColumnsHm to a HashSet or a LinkedHashSet if you want to preserve the order.

Also since it's not possible for partitionedColumnsHm to be null you can avoid null checks against it.

Reply:
> Also since it's not possible for partitionedColumnsHm to be null you can avoid null checks against it.

I think it's possible when there's no partitioning.

Reply:
It's currently initialized as an empty map and never set to null.